components/post_view/embedded_bindings/select_binding/select_binding.tsx

```diff
@@ -43,16 +43,41 @@ type State = {
 
 export class SelectBinding extends React.PureComponent<Props, State> {
     private providers: MenuActionProvider[];
-
+    private nOptions = 0;
     constructor(props: Props) {
         super(props);
 
         const binding = props.binding;
         this.providers = [];
         if (binding.bindings) {
-            const options = binding.bindings.map((b) => {
-                return {text: b.label, value: b.location};
+            const options: Array<{text: string; value: string}> = [];
+            const usedLabels: {[label: string]: boolean} = {};
+            const usedValues: {[label: string]: boolean} = {};
+            binding.bindings.forEach((b) => {
+                const label = b.label || b.location;
+                if (!label) {
+                    return;
+                }
+
+                if (!b.location) {
+                    return;
+                }
+
+                if (usedLabels[label]) {
+                    return;
+                }
+
+                if (usedValues[b.location]) {
+                    return;
+                }
+
+                usedLabels[label] = true;
+                usedValues[b.location] = true;
+
+                options.push({text: label, value: b.location});
             });
```

Comment:
What is the advantage of also checking the semantics here?

Reply:
Not sure what are you asking. Could you elaborate?

Reply:
https://github.com/mattermost/mattermost-webapp/pull/8249#discussion_r662359365 make my point obsolete.